Sources/Extensions/SwiftStdlib/DictionaryExtensions.swift

```diff
@@ -145,7 +145,47 @@ public extension Dictionary where Key: StringProtocol {
             }
         }
     }
+}
+
+// MARK: - Subscripts
+public extension Dictionary {
 
+    /// SwifterSwift: Deep fetch or set a value from nested dictionaries.
+    ///
+    ///        var dict =  ["key": ["key1": ["key2": "value"]]]
+    ///        dict[path: ["key", "key1", "key2"]] = "newValue"
+    ///        dict[path: ["key", "key1", "key2"]] -> "newValue"
+    ///
+    /// - Note: Value fetching is iterative, while setting is recursive.
+    ///
+    /// - Complexity: O(N), _N_ being the length of the path passed in.
+    ///
+    /// - Parameter path: An array of keys to the desired value.
+    ///
+    /// - Returns: The value for the key-path passed in. `nil` if no value is found.
+    public subscript(path path: [Key]) -> Any? {
+        get {
+            return path.reduce(self) { result, key -> Any? in
+                if let element = (result as? [Key: Any])?[key] {
```

Comment:
Using reduce here is more clear, but it makes not possible to an early exit when one of the first keys are not found. 
Example: When dictionary is `["key": ["key1": ["key2": "value"]]]` and we use dic[path: ["keyRandon", "key1", "key2"]] even if it find nil in the first iteration the reduce will continue execute n times. Maybe we can consider a loop implementation here to optimize this case :))